### ISSUE TITLE
Budget allocations lost after app restart in Windows electron app 

### ISSUE THREAD

**Issue by SaveYourMoney** [actualbudget/actual#1263]
### Verified issue does not already exist?

- [X] I have searched and found no existing issue

### What happened?

Had a strange thing happen here over the weekend.  A little background- I have Actual running in a self-hosted environment and I also downloaded the Electron desktop version on Windows 10 to test it as well.  Yesterday morning I sat down and allocated all my income for July to the budget categories I use, doing it first in the self-hosted, and then going over to the desktop  app and mirroring the allocations.  Everything went fine and I closed out of both versions of Actual.

This morning, I opened the desktop app and discovered that all the budget allocations for July were erased and the total amount was showing as available to budget again.  This happened ONLY in the desktop app.  The self-hosted instance had retained my July allocations.  This seems like a possible bug but I don't know how to give further clues as to what may have happened.  To resolve the issue I simply exported from my self-hosted and imported into electron to get back the lost allocations.

Also of note- A single budget transaction for July was NOT lost in the apparent bug.

### What error did you receive?

No error message receive other than the obvious loss of budget allocations previously entered.

### Where are you hosting Actual?

Desktop App (Electron)

### What browsers are you seeing the problem on?

Desktop App (Electron)

### Operating System

Windows 10

**Comment by MatissJanis** [comment]
Might be related to https://github.com/actualbudget/actual/issues/1255

TLDR: the socket connection between UI and the local server sometimes gets lost. At which point no data is stored and/or is retrievable.

**Comment by MatissJanis** [comment]
I believe this is now patched.
